nsqd/gossip_test.go

```diff
@@ -0,0 +1,141 @@
+package nsqd
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+)
+
+var gossipNotifyCh = make(chan struct{}, 20)
+var (
+	converging     = false
+	convergingLock = &sync.Mutex{}
+)
+
+func TestGossip(t *testing.T) {
+	var nsqds []*NSQD
+	var seedNodeAddr string
+	var seedNode *NSQD
+	var tcpPorts []int
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gossipNotify = func() {
+		convergingLock.Lock()
+		defer convergingLock.Unlock()
+		if converging {
+			gossipNotifyCh <- struct{}{}
+		}
+	}
+
+	num := 3
+	for i := 0; i < num; i++ {
+		var nsqd *NSQD
+
+		// find an open port
+		tmpl, err := net.Listen("tcp", hostname+":0")
```

Comment:
basically none of these hostname and other ([recent](https://github.com/bitly/nsq/compare/91a2119...2082caf)) address related changes make sense (for the test specifically)

If we had been explicitly telling it to listen on localhost (setting `GossipAddress`) why is this any different?